src/native/libs/System.Globalization.Native/CMakeLists.txt

```diff
@@ -59,6 +63,18 @@ set(NATIVEGLOBALIZATION_SOURCES
     pal_normalization.c
 )
 
+if (DEFINED CMAKE_ICU_DIR)
+    include_directories(${CMAKE_ICU_DIR}/include)
+    link_libraries(${CMAKE_ICU_DIR}/lib/libicuuc.a ${CMAKE_ICU_DIR}/lib/libicui18n.a ${CMAKE_ICU_DIR}/lib/libicudata.a)
+    link_libraries(stdc++)
```

Comment:
Also https://github.com/dotnet/runtime/pull/82444/files#diff-fbc587d0b04feb58ec4c10941c3beb54321a1c10d0ffb09c44ebee0100a58103R21

Reply:
It always happens somewhere, either at linking the runtime as dynamic library, or as part of the final linking of the application.

In this particular case we happen to build .dylib for iOS which may not be strictly necessary. Since the project is marked as using the "C" language and not "CXX" the C++ standard library is not included implicitly.

We can exclude the .dylib builds as an alternative.

Reply:
Ok, I'm mostly worried about Android where we can't introduce this stdc++ dependency. For Android we're loading ICU .so via dlopen from the system location so we just need to make sure this doesn't apply for Android.

I'd probably move the two `link_libraries` down to the condition where we set STATIC_ICU for iOS/Catalyst etc

Reply:
On Android you will end up linking to the libmono*.so that has the ICU shim built-in. This PR doesn't affect that. We do build an additional libSystem.Globalization.Native.[a/so] and it likely is not necessary. While not necessary for the supported configurations, it is nice to have for the unsupported NativeAOT/Android builds.